### PR TITLE
Dont parse bot comments and allow comments at the end of lgtm cancel

### PR DESCRIPTION
**Pull request by grodrigues3**
Oops, the bot shouldn't be able to approve a PR!

> [APPROVALNOTIFIER] This PR is APPROVED
> 
> The following people have approved this PR: bgrant0607, jessfraz, k8s-merge-robot
> 
> Needs approval from an approver in each of these OWNERS Files:
> 
> OWNERS [bgrant0607]

Though this doesn't affect whether or not the approved label is added, we shouldn't allow this.  So automatically skip comments written by the bot and parse `/lgtm cancel //some comments` so that it's interpreted as a cancel.

### CODE REVIEW

File: mungegithub/mungers/approval-handler.go [hunk 1]
```diff
@@ -371,6 +372,10 @@ func createApproverSet(comments []*githubapi.IssueComment, prAuthor *string) set
 
 	approveComments := getApproveComments(comments)
 	for _, comment := range approveComments {
+		if comment.User.Login != nil && *comment.User.Login == mungerutil.BotName {
```
Comment (apelisse) [inline mungegithub/mungers/approval-handler.go]: Fix the function above: `getApproveComments`. Hint: there is a matcher.

**Comment by apelisse** [comment]
There is an error:
> k8s.io/test-infra/mungegithub/mungers/approval-handler.go:33: can't find import: "k8s.io/test-infra/bazel-test-infra/mungegithub/mungers/mungerutil"

File: mungegithub/mungers/approval-handler.go (hunk 1 above)
Comment (grodrigues3) [inline mungegithub/mungers/approval-handler.go]: you mean fix that line?  cuz the function you refer to already uses matchers

**Comment by grodrigues3** [comment]
Fixed in refactored version

**Pull request closed without merging**